commerce/domain/shipping/cart/calculator/classes/ShippingCartCalculatorSample.cls

```diff
@@ -39,54 +39,57 @@ public class ShippingCartCalculatorSample extends CartExtension.ShippingCartCalc
     } else {
       CartExtension.CartItemList cartItems = cart.getCartItems();
       Integer numberOfUniqueItems = cartItems.size();
-      CartExtension.CartDeliveryGroup cartDeliveryGroup = cartDeliveryGroups.get(0);
-      CartExtension.CartDeliveryGroupMethodList cartDeliveryGroupMethods = cartDeliveryGroup.getCartDeliveryGroupMethods();
-
-      // Clean up the CartDeliveryGroupMethods
-      for (Integer i = (cartDeliveryGroupMethods.size() - 1); i >= 0; i--) {
-        CartExtension.CartDeliveryGroupMethod method = cartDeliveryGroupMethods.get(i);
-        cartDeliveryGroupMethods.remove(method);
-      }
-      // To clear selected Cart Delivery Group Method
-      cartDeliveryGroup.setSelectedCartDeliveryGroupMethod(null);
-
-      // Get the Shipping Product
-      List<Product2> shippingProducts = [SELECT Id FROM Product2 WHERE ProductClass != 'VariationParent' LIMIT 1];
-      if(shippingProducts.size() == 0) {
-        CartExtension.CartValidationOutput cvo = new CartExtension.CartValidationOutput(CartExtension.CartValidationOutputTypeEnum.SHIPPING,
-                CartExtension.CartValidationOutputLevelEnum.ERROR );
-        cvo.setMessage('No Shipping Products have been defined');
-        cartValidationOutputList.add(cvo);
-      } else {
+      
+      for (Integer i = (cartDeliveryGroups.size() - 1); i >= 0; i--) {
```

Comment:
minor comment, you don't have to change just for this:
why not for (int i=0; i < size; i++)? any particular reason?

Reply:
The only thought: cartDeliveryGroups.size() is called just once in the init., but that's really minor

Reply:
and Gurpreet writes cool code, not like everyone )